op-chain-ops/cmd/celo-migrate/state.go

```diff
@@ -113,6 +114,30 @@ func applyStateMigrationChanges(config *genesis.DeployConfig, genesis *core.Gene
 	// uncle blocks, or receipts in the Cel2 transition block.
 	cel2Block := types.NewBlock(cel2Header, nil, nil, nil, trie.NewStackTrie(nil))
 
+	// Create genesis block for new chains to sync
+	genesisBlock := rawdb.ReadBlock(ldb, genesisHash, 0)
+	syncGenesis := core.Genesis{
+		Config:        cfg,
+		Nonce:         genesisBlock.Nonce(),
+		Timestamp:     genesisBlock.Time(),
+		ExtraData:     genesisBlock.Extra(),
+		GasLimit:      genesisBlock.GasLimit(),
+		Difficulty:    genesis.Difficulty,
+		Mixhash:       genesisBlock.MixDigest(),
+		Coinbase:      genesisBlock.Coinbase(),
+		Alloc:         map[common.Address]types.Account{},
```

Comment:
@piersy Any idea if we need to fill those?

Reply:
We will need to fill them since the genesis file needs to setup the correct genesis state. Not sure how we should go about it though.

There is a function `core.ReadGenesis` which gets the completed genesis from the db, but it doesn't work because it is missing the state spec in the database (which is what defines the alloc). Storing allocs in the db was added in [this PR](https://github.com/ethereum/go-ethereum/pull/24460) so I guess we should take the same approach. I.E storing the allocs for the networks we know about.